pandas/tests/io/parser/common/test_ints.py

```diff
@@ -110,6 +116,85 @@ def test_integer_overflow_bug(all_parsers, sep):
     tm.assert_frame_equal(result, expected)
 
 
+def _iinfo(dtype):
+    pdtype = pandas_dtype(dtype)
+    iinfo = np.iinfo(pdtype.type if is_extension_array_dtype(dtype) else pdtype)
+    return iinfo
+
+
+_raises_any_integer_cast_exception = pytest.raises(  # noqa: PDF010
+    (OverflowError, TypeError, ValueError),
+    match="|".join(
+        [
+            "Overflow",
+            "cannot safely cast non-equivalent",
+            "Integer out of range",
+            "Unable to convert column",
+            "The elements provided in the data cannot all be casted to the dtype",
+        ]
+    ),
+)
+
+
+@skip_pyarrow
+@pytest.mark.parametrize(
+    "getval,expected",
+    [
+        (lambda dtype: _iinfo(dtype).max, nullcontext()),  # in range does not raise
+        (lambda dtype: _iinfo(dtype).min, nullcontext()),  # in range does not raise
+        (lambda dtype: _iinfo(dtype).max + 1, _raises_any_integer_cast_exception),
+        (lambda dtype: _iinfo(dtype).min - 1, _raises_any_integer_cast_exception),
+    ],
+)
+def test_integer_overflow_with_user_dtype(all_parsers, any_int_dtype, getval, expected):
```

Comment:
This test is complex. Can you simplify? Even if this would duplicate a bit?

Reply:
I split the test into succeeding max/min vs. failing overflow max+1/min-1 in order to remove the conditional raise (see [f653e96](https://github.com/pandas-dev/pandas/pull/47168/commits/f653e96a43ebeca1399df38fc5563ac108f79d11)). This create little indeed little code duplication and in fact allows for a better name of the test functions.

Note that it seems like we could test the succeeding min/max in a single call to `read_csv` (without pytest parametrization) but this unfortunately fails for `UInt64` (due to intermediate float representation), so I also used `pytest.mark.parametrize` to iterate over min/max.